### PR TITLE
Fix Otlp*MetricExporterBuilderTests

### PR DESCRIPTION
**Pull request by jack-berg**
I ran the build locally for the first time after merging #7255 and noticed a variety of issues:

- Instances where GlobalOpenTelemetry is not reset
- Tests export real data via OTLP without setting up an OTLP receiver, causing error logs
- Test mocks don't account for all the code paths, causing NPEs

Fixing things up here.

### CODE REVIEW

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java [hunk 1]
```diff
@@ -87,9 +87,11 @@ private LongCounter exported() {
   }
 
   private Meter meter() {
-    return meterProviderSupplier
-        .get()
-        .get("io.opentelemetry.exporters." + exporterName + "-" + transportName);
+    MeterProvider meterProvider = meterProviderSupplier.get();
+    if (meterProvider == null) {
+      meterProvider = MeterProvider.noop();
+    }
```
Comment (jack-berg) [inline exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java:93]: This is unrelated, but I noticed that we don't protect against `Supplier<MeterProvider>` returning null. This is one of the edge cases which nullaway doesn't catch.

File: exporters/otlp/all/src/test/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilderTest.java [hunk 2]
```diff
@@ -74,53 +103,61 @@ void setMeterProvider_null() {
   @Test
   void setMeterProvider() {
     when(meterProvider.get(any())).thenReturn(meter);
-    when(meter.counterBuilder(eq("otlp.exporter.seen"))).thenReturn(counterBuilder);
+    when(meter.counterBuilder(any())).thenReturn(counterBuilder);
```
Comment (jack-berg) [inline exporters/otlp/all/src/test/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilderTest.java:106]: ExporterMetrics obtains / increments multiple counters. Only accounting for one of them was causing NPE.

File: exporters/otlp/all/src/test/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilderTest.java [hunk 3]
```diff
@@ -74,53 +103,61 @@ void setMeterProvider_null() {
...
     when(counterBuilder.build()).thenReturn(counter);
 
     try (OtlpHttpMetricExporter exporter =
-        OtlpHttpMetricExporter.builder().setMeterProvider(meterProvider).build()) {
+        OtlpHttpMetricExporter.builder()
+            .setMeterProvider(meterProvider)
+            .setEndpoint("http://localhost:" + server.httpPort() + "/v1/metrics")
```
Comment (jack-berg) [inline exporters/otlp/all/src/test/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilderTest.java:112]: Need to send the data to a real OTLP receiver.

**Review by jaydeluca — APPROVED** [review]

**Review by trask — APPROVED** [review]
(thanks for reviewing @jaydeluca)
